app/helpers/footer_helper.rb

```diff
@@ -82,61 +113,102 @@ def show_authors_and_editors(obj)
     else
-      if num_versions.positive?
-        latest_user = User.safe_find(obj.versions.latest.user_id)
-        if obj.created_at
-          html << :footer_created_by.t(user: user_link(obj.user),
-                                       date: obj.created_at.web_time)
-        end
-        if latest_user && obj.updated_at
-          html << :footer_last_updated_by.t(user: user_link(latest_user),
-                                            date: obj.updated_at.web_time)
-        elsif obj.updated_at
-          html << :footer_last_updated_at.t(date: obj.updated_at.web_time)
-        end
-      else
-        if obj.created_at
-          html << :footer_created_at.t(date: obj.created_at.web_time)
-        end
-        if obj.updated_at
-          html << :footer_last_updated_at.t(date: obj.updated_at.web_time)
-        end
-      end
-      if obj.respond_to?(:num_views) && obj.last_view
-        times = if obj.old_num_views == 1
-                  :one_time.l
-                else
-                  :many_times.l(num: obj.old_num_views)
-                end
-        date = obj.old_last_view&.web_time || :footer_never.l
-        html << :footer_viewed.t(date: date, times: times)
-      end
-      if User.current && obj.respond_to?(:last_viewed_by)
-        time = obj.old_last_viewed_by(User.current)&.web_time || :footer_never.l
-        html << :footer_last_you_viewed.t(date: time)
-      end
+      []
     end
+  end
 
-    # Show RSS log for all of the above.
-    if obj.respond_to?(:rss_log_id) && obj.rss_log_id
-      html << link_to(:show_object.t(type: :log),
-                      activity_log_path(obj.rss_log_id))
+  def html_for_latest_version_or_non_versioned_object(obj, num_versions)
+    html = if num_versions.positive?
+             html_for_latest_version(obj)
+           else
+             html_for_non_versioned_object(obj)
+           end
+
+    html << html_num_views(obj) if obj.respond_to?(:num_views) && obj.last_view
+
+    if User.current && obj.respond_to?(:last_viewed_by)
+      html << html_last_viewed_by(obj)
     end
 
-    html = html.safe_join(safe_br)
-    tag.p(html, class: "small footer-view-stats mt-3")
+    html
+  end
+
+  def html_for_latest_version(obj)
+    latest_user = User.safe_find(obj.versions.latest.user_id)
+    html = html_created_by(obj)
+
+    if latest_user && obj.updated_at
+      html << :footer_last_updated_by.t(user: user_link(latest_user),
+                                        date: obj.updated_at.web_time)
+    elsif obj.updated_at
+      html << :footer_last_updated_at.t(date: obj.updated_at.web_time)
+    end
+
+    html
+  end
+
+  def html_created_by(obj)
+    if obj.created_at
+      [:footer_created_by.t(user: user_link(obj.user),
+                            date: obj.created_at.web_time)]
+    else
+      []
+    end
+  end
+
+  def html_for_non_versioned_object(obj)
+    html = []
+    if obj.created_at
```

Comment:
Can we avoid this conditional? 
See https://github.com/MushroomObserver/mushroom-observer/pull/1510/files#r1197958451